airsenal/scripts/fill_player_table.py

```diff
@@ -80,20 +80,18 @@ def fill_player_table_from_api(season, dbsession):
     dbsession.commit()
 
 
-def make_player_table(seasons=None, dbsession=session):
-    if seasons is None:
+def make_player_table(seasons=[], dbsession=session):
+    if not seasons:
         seasons = [CURRENT_SEASON]
         seasons += get_past_seasons(3)
-
     if CURRENT_SEASON in seasons:
         latest_season = CURRENT_SEASON
     else:
-        latest_season = seasons[-1]
-    seasons.remove(latest_season)
+        latest_season = seasons[0]
 
     make_init_player_table(season=latest_season, dbsession=session)
     make_player_mappings_table(dbsession=session)
-    make_remaining_player_table(seasons=seasons, dbsession=session)
+    make_remaining_player_table(seasons=seasons[1:], dbsession=session)
```

Comment:
Unlikely to happen but this would go wrong if `latest_season = CURRENT_SEASON` but `CURRENT_SEASON` isn't at index 0 in seasons. I think that's what the `seasons.remove(latest_season)` was trying to avoid in the original version.

Reply:
The problem I was getting with `seasons.remove(latest_season)` was that the subsequent functions after `make_player_table()` in `make_init_db()` would have latest_season removed from it because Python passes by reference, so I was avoiding trying to make any chances to season.

Not sure what the best thing to do would be. Maybe a check to see that the seasons are in order? We could use the `season_str_to_year()` function in `airsenal/scripts/scrape_transfermarkt.py` file and just check that the years are in order...

Reply:
Ah right, maybe sort the whole list then? `seasons=list(sorted(seasons, key=season_str_to_year, reverse=True))`, or something like that?

Reply:
Yeah, I think just adding that to the line before `if CURRENT_SEASON in seasons:` would solve this problem.

I don't think it needs the `list()` though as `sorted()` already returns a list object, so I think `seasons=sorted(seasons, key=season_str_to_year, reverse=True)` would be fine.